scripts/check-sample-lnd-conf.sh

```diff
@@ -14,33 +14,45 @@ CONF_FILE=${2:-sample-lnd.conf}
 # We are reading the default values of lnd from lnd --help. To avoid formatting
 # issues the width of the terminal is set to 240. This needs a workaround for
 # CI where we don't have an interactive terminal.
+# LND_OPTIONS is a list of all options of lnd including the equal sign, which
+# is needed to distinguish between booleans and other variables.
+# It is created by reading the first two columns of lnd --help. 
 if [ -t 0 ]; then
     stty cols 240
     LND_HELP="$(go run -tags="$TAGS" \
         github.com/lightningnetwork/lnd/cmd/lnd --help)"" --end"
+    LND_OPTIONS="$(go run -tags="$TAGS" \
+        github.com/lightningnetwork/lnd/cmd/lnd --help | \
+        awk '{
+            option=""; 
+            if ($1 ~ /^--/){option=$1};
+            if ($2 ~ /^--/){option=$2}; 
+            if (match(option,  /--[^=]+[=]*/))
+                {printf "%s ", substr(option, RSTART, RLENGTH)}
+            } 
+            END { printf "%s", "--end"}')"
 else
     tmux new-session -d -s simulated-terminal -x 240 -y 9999
     tmux send-keys -t simulated-terminal.0 "go run -tags=\"$TAGS\" \
         github.com/lightningnetwork/lnd/cmd/lnd --help; tmux wait -S run" ENTER
     tmux wait-for run
     LND_HELP="$(tmux capture-pane -t simulated-terminal.0 -p)"" --end"
-    tmux kill-session -t simulated-terminal
-fi
 
-# LND_OPTIONS is a list of all options of lnd including the equal sign, which
-# is needed to distinguish between booleans and other variables.
-# It is created by reading the first two columns of lnd --help. 
-LND_OPTIONS="$(go run -tags="$TAGS" \
```

Comment:
Shouldn't we instead just change this to `LND_OPTIONS="$(echo "$LND_OPTIONS" | ...`?

Reply:
I think you mean `echo "$LND_HELP"`? I experimented with it at the beginning. But it was pretty hard to make it stable, because the text read from the pane includes also the prompt and the command line flags there.
Imho the script is more stable if we run the awk stuff also in tmux.

Reply:
Then how about instead storing the whole command output in an environment variable and use `tmux capture-pane` we direct the output to a file in the first place? That should reduce the number of commands executed in the if/else flow and prevent the rather long-running command `go run` to only be executed a single time.